src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -248,8 +267,31 @@ public FeatureDataSource(final FeatureInput<T> featureInput, final int queryLook
             this.hasIndex = false;
             this.supportsRandomAccess = true;
         } else if (featureReader instanceof AbstractFeatureReader) {
+
             this.hasIndex = ((AbstractFeatureReader<T, ?>) featureReader).hasIndex();
             this.supportsRandomAccess = hasIndex;
+
+            // Check that our index is not out-of-date:
+            if( hasIndex ) {
```

Comment:
I think this would be best if it's extracted as a method.

Reply:
Fixed!